frame/evm/src/lib.rs

```diff
@@ -735,9 +758,40 @@ impl<T: Config> Pallet<T> {
 			let _ = frame_system::Pallet::<T>::inc_sufficients(&account_id);
 		}
 
+		// Update metadata.
+		let meta = CodeMetadata::from_code(&code);
+		<AccountCodesMetadata<T>>::insert(address, meta);
+
 		<AccountCodes<T>>::insert(address, code);
 	}
 
+	/// Get the account metadata (hash and size) from storage if it exists,
+	/// or compute it from code and store it if it doesn't exist.
+	pub fn account_code_metadata(address: H160) -> CodeMetadata {
+		if let Some(meta) = <AccountCodesMetadata<T>>::get(address) {
+			return meta;
+		}
+
+		let code = <AccountCodes<T>>::get(address);
+
+		// If code is empty we return precomputed hash for empty code.
+		// We don't store it as this address could get code deployed in the future.
+		if code.is_empty() {
+			return CodeMetadata {
+				size: 0,
+				hash: hex_literal::hex!(
+					"c5d2460186f7233c927e7db2dcc703c0e500b653ca82273b7bfad8045d85a470"
+				)
```

Comment:
maybe using a constant here will be better?

Reply:
Do you see any benefit of declaring a constant and using it here?

Reply:
Without reading the comments, it is not easy for me to immediately see what the hex value refers to.
I think the following is better, of course this is just my taste:
```
const EMPTY_CODE_HASH: [u8; 32] = hex_literal::hex!("c5d2460186f7233c927e7db2dcc703c0e500b653ca82273b7bfad8045d85a470");
return CodeMetadata {
    size: 0,
    hash: EMPTY_CODE_HASH.into(),
};
```